Sources/Build/BuildPlan.swift

```diff
@@ -1254,13 +1254,17 @@ public final class ProductBuildDescription {
 
         // Embed the swift stdlib library path inside tests and executables on Darwin.
         if containsSwiftTargets {
+          let useStdlibRpath: Bool
           switch product.type {
-          case .library, .plugin: break
-          case .test, .executable, .snippet:
-              if buildParameters.triple.isDarwin() {
-                  let stdlib = buildParameters.toolchain.macosSwiftStdlib
-                  args += ["-Xlinker", "-rpath", "-Xlinker", stdlib.pathString]
-              }
+          case .library(let type):
+            useStdlibRpath = type == .dynamic
+          case .test, .executable, .snippet, .plugin:
+            useStdlibRpath = true
```

Comment:
I think plugins also need this rpath actually, since they build an executable. It might not have mattered so far, similarly to how it didn't matter that the XCBuild backend and dynamic products didn't pass it, but it seems like this is becoming relevant with the compatibility library for concurrency.

Reply:
Note that the `.plugin` target type only serves to tell SwiftPM about the plugin — the plugin is compiled specially, manifest-style, instead of being produced by the build system.  Plugin targets are similar to binary targets or system library targets in that respect.

Reply:
So I think `.plugin` should be skipped here.